src/main/java/com/epam/task/third/repo/TetrahedronRepositoryImplementation.java

```diff
@@ -0,0 +1,62 @@
+package com.epam.task.third.repo;
+
+import com.epam.task.third.entity.Tetrahedron;
+import com.epam.task.third.repo.specifications.TetrahedronSpecification;
+import com.epam.task.third.validation.RegularTetrahedronValidator;
+import com.epam.task.third.validation.TetrahedronIsNotRegularException;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TetrahedronRepositoryImplementation implements TetrahedronRepository {
+
+    private final Map<Integer, Tetrahedron> repoMap = new HashMap<>();
+    private int id = 0;
+
+    @Override
+    public void addTetrahedron(Tetrahedron tetrahedron) throws TetrahedronIsNotRegularException {
+        if (isRegular(tetrahedron)) {
```

Comment:
just
repoMap.put(id, tetrahedron)
that's it 
no need to do it more complicated
you should validate points before tetrahedron creation so that your app would never have non-regular tetrahedron
what regular means? means invalid?